src/hooks/useMasternodes.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react'
+import { useWhaleApiClient } from '../layouts/contexts/WhaleContext'
+import { masternodes } from '@defichain/whale-api-client'
+
+interface useMasternodesData{
+  loading: boolean
+  masternodes: masternodes.MasternodeData[]
+  hasNext: boolean
+  next: string | number | undefined
+  error: boolean
+}
+
+export function useMasternodes (querySize: number, nextToken?: string): useMasternodesData {
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(false)
+  const [masternodes, setMasternodes] = useState<masternodes.MasternodeData[]>([])
+  const [hasNext, setHasNext] = useState<boolean>(false)
+  const [next, setNext] = useState<string | number | undefined>(undefined)
```

Comment:
we don't need `hasNext, setHasNext` as `next, setNext` can indicate it

Reply:
> i'm thinking about the masternode listing information
> perhaps
> 
> ```
> | masternodeid | status | amount| accumulated profit |
> ```

@canonbrother , i do not not think accumulated profit is possible. Take a look at `MasternodeData`